resources/nersc_supervisor.sh

```diff
@@ -0,0 +1,273 @@
+#  slurm jobs completes by the time we ask what's active.
+# I figure the process itself (and its heartbeat) will manage
+#  the error reporting
+function cleanupLogs {
+  declare -i unfinished
+  logit 2 "cleanupLogs"
+  if [[ ! -d ${NS_SLURM_LOG} ]]
+    then
+      logit 0 "Misconfiguration with slurm log ${NS_SLURM_LOG}"
+      return 1
+    fi
+  #
+  fileList=$(/usr/bin/ls "${NS_SLURM_LOG}/"*.out)
+  if [[ "${fileList}" == "" ]]; then return; fi
+  unfinished=0
+  for filename in ${fileList}
+    do
+       stuff=$(/usr/bin/grep -Ei '^START|^END' "${filename}")
+       bc=$(echo "${stuff}" | /usr/bin/awk 'BEGIN{x=0;}{if($1=="START")x=x+1;if($1=="END")x=x+2;}END{print x;}')
+       if [[ "${bc}" == "3" ]]
+         then
+           if ! /usr/bin/mv "${filename}" "${NS_SLURM_LOG_SEEN}/"
+             then return $?; fi
+         else
+           unfinished=$(( unfinished + 1 ))
+         fi
+    done
+  echo "${unfinished}"
+  return 0
+}
+
+###
+# Check the slurm job activity information, count the incomplete entries
+
+function countIncomplete {
+  #
+  logit 2 'Entering countIncomplete'
+  twoweeksbefore=$(/usr/bin/date +%s | /usr/bin/awk '{printf("%s\n",strftime("%Y-%m-%d",$1));}')
+  listing=$(/usr/bin/sacct -n -b -S "${twoweeksbefore}" -M escori -p | /usr/bin/awk '{split($0,b,"|");split(b[1],a,".");if(b[2]!="COMPLETED")print a[1],b[2],b[3],"|";}' |/usr/bin/sort -u)
+  logit 2 "${listing}"
+  echo "${listing}" | /usr/bin/awk '{n=split($0,a,"|");print n-1;}'
+  return 0
+}
+
+
+
+#####
+# More initialization
+# This is where the main work really begins.
+###
+#
+declare -i NS_LOG_DETAIL count expectedcount fc incomplete activeJobs
+
+export NS_DEBUG="True"
+
+###############################
+# Setup for debugging/testing #
+###############################
+
+if [[ ${NS_DEBUG} == "True" ]]
+ then
+   export NS_FAIL_TEST_SBATCH=0		# Vary these at will
+   export NS_FAIL_TEST_SQUEUE=0
+   export NS_LOG=/scratch/jbellinger/ns.log
+   export NS_SLURM_LOG=/scratch/jbellinger
+   export NS_SLURM_LOG_SEEN=/scratch/jbellinger/seen
+   NS_SCRIPT_PATH=$(pwd)
+   export NS_SCRIPT_PATH
+   export NS_FAKE_QUEUE="234567 bigmem slurm-nersc-mover-234567.out icecubed PD 0:00 1 ()\n 765432 bigmem slurm-nersc-verifier-765432.out icecubed PD 0:00 1 ()"
+   function SBATCH {
+      fakejob=$(date +%s | awk '{print substr($1,4,6);}')
+      fname=$(echo "$3" | /usr/bin/sed "s/%j/${fakejob}/")
+      echo "START" >> "${fname}"
+      echo "END" >> "${fname}"
+      return ${NS_FAIL_TEST_SBATCH}
+   }
+   function SQUEUE {
+      base="CLUSTER: escori\n        JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n"
+      base=${base}"\n"${NS_FAKE_QUEUE}
+      echo "${base}"
+      return ${NS_FAIL_TEST_SQUEUE}
+   }
+ else
+   export NS_LOG=/global/homes/i/icecubed/LTA/ns.log
+   export NS_SLURM_LOG=/global/homes/i/icecubed/LTA/SLURMLOGS
+   export NS_SLURM_LOG_SEEN=/global/homes/i/icecubed/LTA/SLURMLOGS/seen
```

Comment:
For config variables like this, the parent path should be declared as a variable and then used here: 

WORKING_DIR=/global/homes/i/icecubed/LTA
NS_LOG=$WORKING_DIR/ns.log

etc